internal/controller/dnsrecord_controller.go

```diff
@@ -393,38 +447,12 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 	plan = plan.Calculate()
 
 	if err = plan.Error(); err != nil {
-		return noRequeueDuration, err
+		return false, err
 	}
-
-	dnsRecord.Status.ValidFor = defaultRequeueTime.String()
-	dnsRecord.Status.QueuedAt = reconcileStart
 	if plan.Changes.HasChanges() {
-		// generation has not changed but there are changes.
-		// implies that they were overridden - bump write counter
-		if !generationChanged(dnsRecord) {
-			if dnsRecord.Status.WriteCounter < WriteCounterLimit {
-				dnsRecord.Status.WriteCounter++
-				wrtiteCounter.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace).Inc()
-				logger.V(1).Info("Changes needed on the same generation of record")
-			} else {
-				err = errors.New("reached write limit to the DNS provider for the same generation of record")
-				logger.Error(err, "Giving up on trying to maintain desired state of the DNS record - changes are being overridden")
-				return noRequeueDuration, err
-			}
-		}
-		dnsRecord.Status.ValidFor = validationRequeueTime.String()
-		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, "AwaitingValidation", "Awaiting validation")
 		logger.Info("Applying changes")
 		err = registry.ApplyChanges(ctx, plan.Changes)
-		if err != nil {
-			return validationRequeueTime, err
-		}
-	} else {
-		logger.Info("All records are already up to date")
-		dnsRecord.Status.WriteCounter = 0
-		wrtiteCounter.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace).Set(0)
-		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionTrue, "ProviderSuccess", "Provider ensured the dns record")
+		return true, err
 	}
-
-	return defaultRequeueTime, nil
+	return false, nil
```

Comment:
The boolean returned here will mean there were changes and they were successfully, iF the plan has changes but they aren't applied it will return false and we rely on the error wherever we are needing to update status right?

Can you add a description to make it clear what the purpose of this boolean value is and what the value is expected to be when there are calculated changes but they can't be applied for whatever reason.

Reply:
Not exactly. If the plan had changes we will return `true`. The success or failure in applying will affect only the `err` returned. I've added more comments around it to make it less confusing.

Reply:
OK